dams-key-server/src/command.rs

```diff
@@ -1,157 +1,13 @@
-                                    rng,
-                                    &client,
-                                    &config,
-                                    &service,
-                                    session_key,
-                                    chan,
-                                ).await?,
-                                1 => Register.run(
-                                    db,
-                                    &mut rng,
-                                    &client,
-                                    &config,
-                                    &service,
-                                    session_key,
-                                    chan,
-                                ).await?,
-                                2 => Retrieve.run(
-                                    rng,
-                                    &client,
-                                    &config,
-                                    &service,
-                                    session_key,
-                                    chan,
-                                ).await?,
-                                3 => Authenticate.run(
-                                    db,
-                                    &mut rng,
-                                    &client,
-                                    &config,
-                                    &service,
-                                    session_key,
-                                    chan,
-                                ).await?,
-                            })?;
-                            Ok::<_, anyhow::Error>(())
-                        }
-                    };
-
-                    // Future that completes on graceful shutdown
-                    let wait_terminate = async move { wait_terminate.recv().await.unwrap_or(()) };
-
-                    // Run the server until graceful shutdown
-                    server
-                        .serve_while(
-                            address,
-                            Some((&certificate, &private_key)),
-                            initialize,
-                            interact,
-                            wait_terminate,
-                            |address| info!("{}", TestLogs::ServerSpawned(address)),
-                        )
-                        .await?;
-                    Ok::<_, anyhow::Error>(())
-                }
-            })
-            .collect();
-
-        // Wait for the server to finish
-        tokio::select! {
-            _ = signal::ctrl_c() => info!("Terminated by user"),
-            Some(Err(e)) = server_futures.next() => {
-                error!("Error: {}", e);
-            },
-            else => {
-                info!("Shutting down...")
-            }
-        }
-
-        Ok(())
-    }
+use dams::user::UserId;
+use std::str::FromStr;
+use tonic::Status;
+
+pub mod authenticate;
+pub mod register;
+
+fn user_id_from_message(message: &[u8]) -> Result<UserId, Status> {
```

Comment:
```suggestion
pub(crate) fn user_id_from_message(message: &[u8]) -> Result<UserId, Status> {
```

Nitpick: Rust allows this but it looks a little confusing as a reader.